### PR TITLE
avoid downstream users from relying on serialization to create a copy of the binding

### PR DESCRIPTION
**Pull request by KristofferC**
See analyis in https://github.com/JuliaLang/julia/issues/48078#issuecomment-1371885752.

### CODE REVIEW

File: docs/src/extending.md [hunk 1]
```diff
@@ -41,7 +41,7 @@ In the unlikely case that you've used `@dimension`, you will also need the
 following incantation:
 
 ```julia
-const localpromotion = Unitful.promotion
+const localpromotion = copy(Unitful.promotion)
 function __init__()
     Unitful.register(YourModule)
     merge!(Unitful.promotion, localpromotion)
```
[the remark this answers is not in the document]
Comment (KristofferC) [reply docs/src/extending.md:47]: Yeah, I thought about something a little bit similar but felt that the simplicity of this was ok for now while a better method might be developed to fully replace this.

[the remark this answers is not in the document]
Comment (sostock) [reply docs/src/extending.md:47]: I also think this is fine for now, but we should aim to replace this mechanism in a future release.

@giordano What do you think?

[the remark this answers is not in the document]
Comment (giordano) [reply docs/src/extending.md:47]: Ok, sounds good, but we should have an issue about how to improve this in the future.

[the remark this answers is not in the document]
Comment (sostock) [reply docs/src/extending.md:47]: Issue created: #613

**Pull request merged**